Avoid ioutil package

mark-rushakoff opened the pull request:
As of go1.16, the ioutil package's functionality is provided by packages
io and os. This commit changes to use those implementations, except for
the case of ioutil.ReadDir; the API of os.ReadDir is slightly different,
so that will be changed in a separate commit.

mark-rushakoff committed:
Avoid ioutil package

As of go1.16, the ioutil package's functionality is provided by packages
io and os. This commit changes to use those implementations, except for
the case of ioutil.ReadDir; the API of os.ReadDir is slightly different,
so that will be changed in a separate commit.

jtieri reviewed (changes requested):
Looks like the import statement for the `os` package is missing so tests are failing.
Other than that LGTM

mark-rushakoff committed:
Merge remote-tracking branch 'origin/main' into chore/avoid-ioutil

mark-rushakoff committed:
Use go1.17 in github workflows

The current go.mod specifies go1.17, but the CI jobs were configured to
use go1.15.
